components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/AssetBlob.java

```diff
@@ -35,6 +36,8 @@
 {
   private final BlobRef blobRef;
 
+  private final Blob blob;
```

Comment:
There's something odd about storing both `BlobRef` and `Blob`... does it make sense to tweak the BlobStore API so that we can get everything we need here from `Blob`? Or is it a design decision that `BlobRef` contains additional information that's not available from `Blob`?